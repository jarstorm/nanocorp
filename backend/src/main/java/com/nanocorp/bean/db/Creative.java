package com.nanocorp.bean.db;

import java.io.Serializable;

import javax.persistence.*;

@Entity
@Table(name = "NC_CREATIVE")
public class Creative implements Serializable {

	private static final long serialVersionUID = -5293108294821693505L;

	/**
	 * Creative id
	 */
	@Id
	@Column(name = "C_ID")
	@GeneratedValue(strategy = GenerationType.AUTO)
	private long id;

	@Column(name = "C_HEADER")
	@Lob
	private String header;

	@Column(name = "C_HEADER_1")
	@Lob
	private String header1;

	@Column(name = "C_HEADER_2")
	@Lob
	private String header2;

	@Column(name = "C_DESCRIPTION")
	@Lob
	private String description;

	@Column(name = "C_URL")
	@Lob
	private String url;

	@Column(name = "C_IMAGE")
	@Lob
	private String image;

	@OneToOne
	@PrimaryKeyJoinColumn
	private Platform platform;

	public long getId() {
		return id;
	}

	public void setId(long id) {
		this.id = id;
	}

	public String getHeader() {
		return header;
	}

	public void setHeader(String header) {
		this.header = header;
	}

	public String getHeader1() {
		return header1;
	}

	public void setHeader1(String header1) {
		this.header1 = header1;
	}

	public String getHeader2() {
		return header2;
	}

	public void setHeader2(String header2) {
		this.header2 = header2;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public String getUrl() {
		return url;
	}

	public void setUrl(String url) {
		this.url = url;
	}

	public String getImage() {
		return image;
	}

	public void setImage(String image) {
		this.image = image;
	}

}
