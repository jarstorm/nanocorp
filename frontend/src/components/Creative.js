import React, {Component} from 'react';
import {SERVER_URL} from '../../config.js';

export default class Creative extends Component {
  	render() {
	  	const {data} = this.props;
	  	if (data !== null) {
		    return (	    		
		      <div className="col-md-4">	
		      	<p><b>Creative</b></p>      
		      	<p>Description: {data.description}</p>
		      	<p>Header: {data.header}</p>
		      	<p>Header1: {data.header1}</p>
		      	<p>Header2: {data.header2}</p>
		      	<p>Url: {data.url}</p>
		      	<p>Image: </p><img src={SERVER_URL + "/image/" + data.image} className="creative-img"/>
		      </div>
		    )
		} else {
			return (<div><p><b>Creative</b></p></div>)
		}
  	}
}
