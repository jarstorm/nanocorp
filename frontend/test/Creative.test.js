// Link.react.test.js
import React from 'react';
import Creative from '../src/components/Creative';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

test('Draw Creative component', () => {  
  const data = {description: 'description', header1: 'header1', header2: 'header2',  url: 'url', image: 'image'}
  const component = renderer.create(<Creative data={data} />,
  );
  let dialog = component.toJSON();
  expect(dialog).toMatchSnapshot();  
});
