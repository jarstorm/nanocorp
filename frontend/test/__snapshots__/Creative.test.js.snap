// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Draw Creative component 1`] = `
<div>
  <p>
    <b>
      Creative
    </b>
  </p>
  <p>
    Description: 
    description
  </p>
  <p>
    Header: 
  </p>
  <p>
    Header1: 
    header1
  </p>
  <p>
    Header2: 
    header2
  </p>
  <p>
    Image: 
  </p>
  <img
    src="http://localhost:8090/images/image"
  />
  <p>
    Url: 
    url
  </p>
</div>
`;
